web/src/main/java/com/microsoft/applicationinsights/web/internal/httputils/HttpServerHandler.java

```diff
@@ -0,0 +1,130 @@
+     * {@link com.microsoft.applicationinsights.web.extensibility.modules.WebRequestTrackingTelemetryModule#onBeginRequest(HttpServletRequest, HttpServletResponse)}
+     * @param request incoming Request
+     * @param response Response object
+     * @return {@link RequestTelemetryContext} that contains correlation information and metadata about request
+     * @throws MalformedURLException
+     */
+    public RequestTelemetryContext handleStart(P request, Q response) throws MalformedURLException {
+        RequestTelemetryContext context = new RequestTelemetryContext(new Date().getTime(),null);
+        RequestTelemetry requestTelemetry = context.getHttpRequestTelemetry();
+        ThreadContext.setRequestTelemetryContext(context);
+
+        String method = extractor.getMethod(request);
+        String userAgent = extractor.getUserAgent(request);
+
+        String uriWithoutSessionId = extractor.getURI(request);
+        String scheme = extractor.getScheme(request);
+        String host = extractor.getHost(request);
+        String query = extractor.getQuery(request);
+
+        if (!CommonUtils.isNullOrEmpty(query)) {
+            requestTelemetry.setUrl(scheme + "://" + host + uriWithoutSessionId + "?" + query);
+        } else {
+            requestTelemetry.setUrl(scheme + "://" + host + uriWithoutSessionId);
+        }
+
+        requestTelemetry.setHttpMethod(method);
+
+        requestTelemetry.setName(method + " " + uriWithoutSessionId);
+        requestTelemetry.getContext().getUser().setUserAgent(userAgent);
+        requestTelemetry.setTimestamp(new Date(context.getRequestStartTimeTicks()));
+
+        webModulesContainer.invokeOnBeginRequest(request, response);
+
+        return context;
+    }
+
+    /**
+     * This method is used to indicate request end instrumentation, complete correlation and record timing, response
+     * @param request HttpRequest object
+     * @param response HttpResponse object
+     */
+    public void handleEnd(P request, Q response) {
+        RequestTelemetryContext context = ThreadContext.getRequestTelemetryContext();
+        RequestTelemetry requestTelemetry = context.getHttpRequestTelemetry();
+        long endTime = new Date().getTime();
+        requestTelemetry.setDuration(new Duration(endTime - context.getRequestStartTimeTicks()));
+        int resultCode = extractor.getStatusCode(response);
+        requestTelemetry.setSuccess(resultCode < 400);
+        requestTelemetry.setResponseCode(Integer.toString(resultCode));
+        webModulesContainer.invokeOnEndRequest(request, response);
+    }
+
+    /**
+     * This method is used to capture runtime exceptions while processing request
+     * @param e Exception occurred
+     */
+    public void handleException(Exception e) {
+        try {
+            InternalLogger.INSTANCE.trace("Unhandled application exception: %s", ExceptionUtils.getStackTrace(e));
```

Comment:
Minor, is this "application exception" or general "exception"?
The comment also mentioned "runtime exception", which confused me since Java has **RuntimeException**.

Reply:
To be specific, this handles. - IOException, ServletException and RuntimeExceptions thrown while processing the request.

Reply:
I updated the log message to be more clear.